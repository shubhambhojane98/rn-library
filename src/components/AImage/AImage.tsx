import React, {FC} from 'react';
import {Image} from 'react-native';
import FastImage from 'react-native-fast-image';
import {moderateScale} from 'react-native-size-matters';
import {defaultScale} from '../../utils/Common';
import {withTheme, useTheme} from '../../core/theming';
import type {Theme} from '../../utils/types';

interface Prop {
  source?: object;
  sourceUri?: string;
  width?: number;
  height?: number;
  backgroundColor?: string;
  resizeMode?: any;
  marginTop?: number;
  marginRight?: number;
  marginBottom?: number;
  marginLeft?: number;
  cache?: any;
  borderRadius?: number;
  theme: Theme;
}

const defaultSize = 64;
const defaultRadius = 0;
const defaultMargin = 0;

const AImage: FC<Prop> = ({
  source,
  sourceUri,
  width,
  height,
  backgroundColor,
  resizeMode,
  marginTop,
  marginRight,
  marginBottom,
  marginLeft,
  cache,
  borderRadius,
}) => {
  const {colors} = useTheme();
  return (
    <>
      {sourceUri && (
        <FastImage
          style={{
            width: moderateScale(width || defaultSize, defaultScale),
            height: moderateScale(height || defaultSize, defaultScale),
            backgroundColor: backgroundColor || colors.transparent,
            marginTop: moderateScale(marginTop || defaultMargin, defaultScale),
            marginRight: moderateScale(
              marginRight || defaultMargin,
              defaultScale,
            ),
            marginBottom: moderateScale(
              marginBottom || defaultMargin,
              defaultScale,
            ),
            marginLeft: moderateScale(
              marginLeft || defaultMargin,
              defaultScale,
            ),
            borderRadius: borderRadius || defaultRadius / 2,
          }}
          source={{
            uri: sourceUri,
            priority: FastImage.priority.normal,
            cache: cache,
          }}
          resizeMode={resizeMode}
        />
      )}
      {source && (
        <Image
          source={source}
          style={{
            width: moderateScale(width || defaultSize, defaultScale),
            height: moderateScale(height || defaultSize, defaultScale),
            backgroundColor: backgroundColor || colors.transparent,
            marginTop: moderateScale(marginTop || defaultMargin, defaultScale),
            marginRight: moderateScale(
              marginRight || defaultMargin,
              defaultScale,
            ),
            marginBottom: moderateScale(
              marginBottom || defaultMargin,
              defaultScale,
            ),
            marginLeft: moderateScale(
              marginLeft || defaultMargin,
              defaultScale,
            ),
            borderRadius: borderRadius || defaultRadius / 2,
          }}
          resizeMode={resizeMode}
        />
      )}
    </>
  );
};

export default withTheme(AImage);
