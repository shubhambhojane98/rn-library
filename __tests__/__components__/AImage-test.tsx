import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import AImage from '../../src/components/AImage/AImage';

it('renders correctly with image URL', () => {
  const tree = renderer
    .create(<AImage sourceUri="https://unsplash.it/400/400?image=1" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// it('renders correctly with local image path', () => {
//   const tree = renderer
//     .create(<AImage source={require('../../assets/images/FullStar.png')} />)
//     .toJSON();
//   expect(tree).toMatchSnapshot();
// });
