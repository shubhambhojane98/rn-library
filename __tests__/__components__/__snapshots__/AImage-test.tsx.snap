// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with image URL 1`] = `
<View
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "backgroundColor": "transparent",
        "borderRadius": 0,
        "height": 78.62857142857143,
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "width": 78.62857142857143,
      },
    ]
  }
>
  <FastImageView
    resizeMode="cover"
    source={
      Object {
        "cache": undefined,
        "priority": "normal",
        "uri": "https://unsplash.it/400/400?image=1",
      }
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</View>
`;
